src/Collapse.js

```diff
@@ -92,11 +90,25 @@ class Collapse extends Component {
       navbar,
       cssModule,
       children,
-      ...transitionProps
+      ...otherProps
     } = this.props;
-    const otherProps = omit(transitionProps, Object.keys(propTypes));
+
     const { height } = this.state;
 
+    // In NODE_ENV=production the Transition.propTypes are wrapped which results in an
+    // empty object "{}". This is the result of the `react-transition-group` babel
+    // configuration settings. Therefore, to ensure that production builds work without
+    // error, we can either explicitly define keys or use the Transition.defaultProps.
+    // Using the Transition.defaultProps excludes any required props. Thus, the best
+    // solution is to explicitly define required props in our utilities and reference these.
+    // This also gives us more flexibility in the future to remove the prop-types
+    // dependency in distribution builds (Similar to how `react-transition-group` does).
+    // Note: Without omitting the `react-transition-group` props, the resulting child
+    // Tag component would inherit the Transition properties as attributes for the HTML
+    // element which results in errors/warnings for non-valid attributes.
+    const transitionProps = pick(otherProps, TransitionPropTypeKeys);
+    const childProps = omit(otherProps, TransitionPropTypeKeys);
+
     return (
       <Transition
```

Comment:
I would like to have a basic fallback which instantly shows/hides (no animation), but we would probably want to tie in the callbacks to that to be more compatible.

Reply:
At some point in future we can do better fallback whereby we don't even use the Fade elements for non fading elements. A lot more work needs to be done for that.